Update the Create User API docs to include Optional Password scenarios (#3188)
okta/okta-developer-docs: packages/@okta/vuepress-site/docs/reference/api/users/index.md

| X                |            | FALSE                      | `STAGED`        |                          |                  |
| X                |            | TRUE                       | `PROVISIONED` or `ACTIVE`   | One-Time Token (Email) or Email  | X                |
|                  | X          | FALSE                      | `STAGED`        |                          |                  |
|                  | X          | TRUE                       | `ACTIVE`        | Password                 | X                |
| X                | X          | FALSE                      | `STAGED`        |                          |                  |
| X                | X          | TRUE                       | `ACTIVE`        | Password                 |                  |

Creating users with a `FEDERATION` or `SOCIAL` provider sets the user status to either `ACTIVE` or `STAGED` based on the `activate` query parameter since these two providers don't support a `password` or `recovery_question` credential.

#### Create User with Optional Password enabled

<ApiLifeCycle access="ea" />

> When Optional Password is enabled, user status following creation can be affected by enrollment policy (see [About MFA enrollment policies and rules](/Content/Topics/identity-engine/policies/about-mfa-enrollment-policies.htm)).

Author: I don't know how to input the link in this case, because I don't think its part of the API reference repo.
The full link is: https://help.okta.com/oie/en-us/Content/Topics/identity-engine/policies/create-mfa-policy.htm.
Should I use that?

> When Optional Password is enabled, user status following creation can be affected by enrollment policy (see [About MFA enrollment policies and rules](/Content/Topics/identity-engine/policies/about-mfa-enrollment-policies.htm)).

Reviewer: ```suggestion
When Optional Password is enabled, the user status following user creation can be affected by the enrollment policy. See [About MFA enrollment policies and rules](https://help.okta.com/okta_help.htm?type=oie&id=ext-create-mfa-policy).
```

> Based on the group memberships that are specified when the user is created (see [Create user in a group](#Create User in Group)) a password may or may not be required to make the users status `ACTIVE`.

Reviewer: ```suggestion
Based on the group memberships that are specified when the user is created, a password may or may not be required to make the users status `ACTIVE`. See [Create user in a group](#Create User in Group).
```

> Based on the group memberships that are specified when the user is created, a password may or may not be required to make the users status `ACTIVE`. See [Create user in a group](#Create User in Group).

Reviewer: ```suggestion
Based on the group memberships that are specified when the user is created, a password may or may not be required to make the user's status `ACTIVE`. See [Create user in a group](#Create User in Group).
```

> If the enrollment policy that applies to the user (as determined by the groups assigned to the user) specifies that Password authenticator is `required` then, in the case where the user is created without a password, the user will be in the `PROVISIONED` state and

Reviewer: ```suggestion
If the enrollment policy that applies to the user (as determined by the groups assigned to the user) specifies that the Password authenticator is `required`, then in the case where the user is created without a password, the user is in the `PROVISIONED` state and
```

> a One-Time Token will be sent to the user via email.

Reviewer: ```suggestion
a One-Time Token is sent to the user through email.
```

> If the user is created with a password, then their state is set to ACTIVE, and they can immediately log in using their Password authenticator.

Reviewer: ```suggestion
If the user is created with a password, then their state is set to ACTIVE, and they can immediately sign in using their Password authenticator.
```

> On the other hand, if the enrollment policy that applies to the groups specified for the newly created user indicates that password is `optional` or `disabled` then the Administrator cannot specify a password for the user.  Instead, the user status is set to `ACTIVE` and the user may immediately log in using their Email authenticator.   If policy permits, and the user so chooses, they can enroll a password after they log in.

Reviewer: ```suggestion
If the enrollment policy that applies to the groups specified for the newly created user indicates that password is `optional` or `disabled`, then the Administrator can't specify a password for the user.  Instead, the user status is set to `ACTIVE` and the user may immediately sign in using their Email authenticator.  If policy permits, and the user so chooses, they can enroll a password after they sign in.
```

Users who don't have a password must complete the welcome flow by visiting the activation link to complete the transition to `ACTIVE` status.

> >**Note:** If you have Optional Password enabled, visiting the activation link is optional for users who are not required to enroll a password (see [Create user with Optional Password enabled](####Create user with Optional Password enabled)).

Reviewer: ```suggestion
> **Note:** If you have Optional Password enabled, visiting the activation link is optional for users who aren't required to enroll a password. See [Create user with Optional Password enabled](#create-user-with-optional-password-enabled).
```
